arelle/plugin/validate/ESEF/Dimensions.py

```diff
@@ -18,28 +19,40 @@
 try:
     import regex as re
 except ImportError:
-    import re
+    import re  # type: ignore[no-redef]
+from collections.abc import Callable
+from arelle.ValidateXbrl import ValidateXbrl
+from arelle.ModelDtsObject import ModelConcept
+
+_: Callable[[str], str]  # Handle gettext
 
-def checkFilingDimensions(val):
+
+def checkFilingDimensions(val: ValidateXbrl) -> None:
 
     val.primaryItems = set() # concepts which are line items (should not also be dimension members
-    val.domainMembers = set() # concepts which are dimension domain members
+    val.domainMembers = set()  # concepts which are dimension domain members
+
     elrPrimaryItems = defaultdict(set)
-    hcPrimaryItems = set()
-    hcMembers = set()
-    
-    def addDomMbrs(sourceDomMbr, ELR, membersSet):
+    hcPrimaryItems: set[ModelConcept] = set()
+    hcMembers: set[Any] = set()
+
+    def addDomMbrs(sourceDomMbr: ModelConcept, ELR: str, membersSet: set[ModelConcept]) -> None:
         if isinstance(sourceDomMbr, ModelConcept) and sourceDomMbr not in membersSet:
             membersSet.add(sourceDomMbr)
             for domMbrRel in val.modelXbrl.relationshipSet(XbrlConst.domainMember, ELR).fromModelObject(sourceDomMbr):
                 #if domMbrRel.isUsable:
                 addDomMbrs(domMbrRel.toModelObject, domMbrRel.consecutiveLinkrole, membersSet)
-            
+
     for hasHypercubeArcrole in (XbrlConst.all, XbrlConst.notAll):
         hasHypercubeRelationships = val.modelXbrl.relationshipSet(hasHypercubeArcrole).fromModelObjects()
+
+        # fromModelObjects() has an optional return type, we need to handle that here
+        if hasHypercubeRelationships is None:
+            continue
```

Comment:
I'll revert this in a follow-up, we agreed that `hasHypercubeRelationships` cannot be `None`.